src/utils/cart-checkout.js

```diff
@@ -0,0 +1,14 @@
+const calculateDiscount = (cart) => cart.reduce(
+          (acc, curr) =>
+            acc +
+            Number(curr.qty) *
+              (Number(curr.orig_price) - Number(curr.curr_price)),
+          0
+        )
+
+const calculatePrice = (cart) =>  cart.reduce(
+          (acc, curr) => acc + Number(curr.qty) * Number(curr.orig_price),
+          0
+        )
+        
+export {calculateDiscount,calculatePrice}
```

Comment:
Nicely done. Same could have been done for cartSize too? or is there anything that I am missing here?

Reply:
Size is controlled in cartreducer and it's required to only change by 1 for add,increment,decrement so I haven't written array reducer for this. For remove item in cart i have used though but it's in the cart reducer action itself